test/performance/odp_bench_misc.c

```diff
@@ -103,6 +113,12 @@ static void init_time_global(void)
 	for (i = 0; i < REPEAT_COUNT; i++)
 		t2[i] = odp_time_global();
 
+	for (i = 0; i < REPEAT_COUNT; i++)
+		t4[i] = odp_time_global_from_ns(ODP_TIME_MSEC_IN_NS);
```

Comment:
I suggest having a separate init-functions for the time-from-ns tests so that the init function of other tests is not affected. Now the added array init is done for all tests that share the init function, even though they do not use the added t4 and t5 arrays. And since the init function is run after the measurement warmup, any side effects (like cache evictions of the actually used data) can affect the measurements. With separate init functions the actual test function could be exactly the same for the two added tests, which would be useful in making sure that the "short" and "long" tests test exactly the same thing.

The test warmup probably should be improved by having a test specific warmup round after the init phase, but that requires changes in the common bench API. In the mean time having a separate init and not introducing t4 and t5 would minimize the effect of this change to existing tests.

The warmup implementation in the common bench framework is also otherwise problematic: it runs all warmups before any test runs, which is too early. PR 1959 fixes that but leaves the init-after-warmup problem as is.

Reply:
Changed in v2 initialization to happen only once in main thread startup.